new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMap.java

```diff
@@ -0,0 +1,603 @@
+        .metadata(new V1ObjectMeta()
+            .name(encryptionSecretName)
+            .namespace(domainNamespace))
+        .stringData(encryptionSecretMap)), "Create secret failed with ApiException");
+    assertTrue(secretCreated, String.format("create secret failed for %s", encryptionSecretName));
+
+    // Create a ConfigMap with a sparse jdbc model file
+    Map<String, String> labels = new HashMap<>();
+    labels.put("weblogic.domainUid", domainUid);
+    String dsModelFile = MODEL_DIR + "/model.jdbc.yaml";
+    Map<String, String> data = new HashMap<>();
+    String configMapName = "dsconfigmap";
+    String cmData = null;
+    cmData  = assertDoesNotThrow(() -> Files.readString(Paths.get(dsModelFile)),
+        String.format("readString operation failed for %s", dsModelFile));
+    assertNotNull(cmData, String.format("readString() operation failed while creating ConfigMap %s", configMapName));
+    data.put("model.jdbc.yaml",cmData);
+
+    V1ObjectMeta meta = new V1ObjectMetaBuilder()
+        .withLabels(labels)
+        .withName(configMapName)
+        .withNamespace(domainNamespace)
+        .build();
+    V1ConfigMap configMap = new V1ConfigMapBuilder()
+        .withData(data)
+        .withMetadata(meta)
+        .build();
+
+    boolean cmCreated = assertDoesNotThrow(() -> createConfigMap(configMap),
+        String.format("createConfigMap failed for %s", configMapName));
+    assertTrue(cmCreated, String.format("createConfigMap failed while creating ConfigMap %s", configMapName));
+
+    // Create the domain custom resource with a predefined configmap 
+    // in configuration/model section
+    Domain domain = new Domain()
+        .apiVersion(API_VERSION)
+        .kind("Domain")
+        .metadata(new V1ObjectMeta()
+            .name(domainUid)
+            .namespace(domainNamespace))
+        .spec(new DomainSpec()
+            .domainUid(domainUid)
+            .domainHomeSourceType("FromModel")
+            .image(miiImage)
+            .addImagePullSecretsItem(new V1LocalObjectReference()
+                .name(REPO_SECRET_NAME))
+            .webLogicCredentialsSecret(new V1SecretReference()
+                .name(adminSecretName)
+                .namespace(domainNamespace))
+            .includeServerOutInPodLog(true)
+            .serverStartPolicy("IF_NEEDED")
+            .serverPod(new ServerPod()
+                .addEnvItem(new V1EnvVar()
+                    .name("JAVA_OPTIONS")
+                    .value("-Dweblogic.StdoutDebugEnabled=false"))
+                .addEnvItem(new V1EnvVar()
+                    .name("USER_MEM_ARGS")
+                    .value("-Djava.security.egd=file:/dev/./urandom ")))
+            .adminServer(new AdminServer()
+                .serverStartState("RUNNING")
+                .adminService(new AdminService()
+                    .addChannelsItem(new Channel()
+                        .channelName("default")
+                        .nodePort(30711))))
```

Comment:
set '0'

Reply:
using hard coded node port does not work if the port is already in use.

Reply:
Added a method getAdminServiceNodePort() to get the Admin Service NodePort.  Now the test set ZERO as node port value while creating the domain resource.